optaplanner-examples/src/main/java/org/optaplanner/examples/batchscheduling/solver/score/BatchSchedulingEasyScoreCalculator.java

```diff
@@ -0,0 +1,244 @@
+package org.optaplanner.examples.batchscheduling.solver.score;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore;
+import org.optaplanner.core.api.score.calculator.EasyScoreCalculator;
+import org.optaplanner.examples.batchscheduling.domain.Allocation;
+import org.optaplanner.examples.batchscheduling.domain.AllocationPath;
+import org.optaplanner.examples.batchscheduling.domain.Schedule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BatchSchedulingEasyScoreCalculator implements EasyScoreCalculator<Schedule, BendableLongScore> {
+
+    final Logger logger = LoggerFactory.getLogger(BatchSchedulingEasyScoreCalculator.class);
+    public static final String ROUTE_PATH_SEPERATOR = "---";
+    private static final int OTHER_ALLOCATION_PENALTY = 1;
+    private static final int CURRENT_ALLOCATION_PENALTY = 1;
+    private static final boolean COMPUTE_hard1Score = true;
+    private static final boolean COMPUTE_soft2Score = true;
+
+    public BendableLongScore calculateScore(Schedule schedule) {
```

Comment:
This method definitely needs to be broken down into smaller parts, it is too long.
Also see my suggestion below.